l3build-check.lua

```diff
@@ -583,13 +592,15 @@ function runcheck(name, hide)
   local errorlevel = 0
   for _,engine in pairs(checkengines) do
     setup_check(name,engine)
-    local errlevel = 0
-    if fileexists(testfiledir .. "/" .. name .. pvtext) then
-      errlevel = check_and_diff(pvtext,engine,compare_pdf,true)
-    else
-      errlevel = check_and_diff(lvtext,engine,compare_tlg)
+    local errlevel
+    for _, kind in ipairs(test_order) do
+      local test_type = test_types[kind]
+      if test_filename == testfiledir .. "/" .. name .. test_type.test then
+        errlevel = check_and_diff(test_type, engine)
+        break
+      end
     end
-    if errlevel ~= 0 and options["halt-on-error"] then
+    if assert(errlevel) ~= 0 and options["halt-on-error"] then
```

Comment:
Do we really want an assertion here? Won't that leave us with an uncontrolled error?

Reply:
It doesn't do anything because `errlevel` can never be `nil` here. (I had the assert to ensure that I didn't miss a case where it can happen, but I'll take it out again.